coltypes/coltypes.go

```diff
@@ -52,24 +54,32 @@ func TransformRow(row map[string]any, columnTypes map[string]ColumnData) (map[st
 			continue
 		}
 
-		switch data := columnTypes[key]; {
-		case data.Type == numberType:
-			if data.Precision != nil && *data.Precision == 1 && data.Scale != nil && *data.Scale == 0 {
-				if v, ok := value.(int64); ok {
-					if v == 1 {
-						result[key] = true
-					} else {
-						result[key] = false
-					}
-
-					continue
-				}
+		if columnTypes[key].Type == numberType {
+			valStr, err := godror.Num.ConvertValue(value)
```

Comment:
is valStr has string type?

Reply:
No, it's an interface.
I know it looks not really cool, but I haven't find a better way to convert Oracles' types :(

Reply:
for me it is looks strange name. valStr  sounds like it  is string value, could you please rename it

Reply:
Oh, didn't get you. Will change.

Reply:
Done.